File: app/src/main/java/com/example/josycom/flowoverstack/viewmodel/AnswerViewModel.java
package com.example.josycom.flowoverstack.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.josycom.flowoverstack.model.Answer;
import com.example.josycom.flowoverstack.repository.AnswerRepository;

import java.util.List;

public class AnswerViewModel extends ViewModel {

    private LiveData<List<Answer>> mAnswersLiveData;

    AnswerViewModel(int questionId, String order, String sortCondition, String site, String filter) {
        AnswerRepository answerRepository = new AnswerRepository(questionId, order, sortCondition, site, filter);
        mAnswersLiveData = answerRepository.getAnswers();
    }

    public LiveData<List<Answer>> getAnswersLiveData() {
        return mAnswersLiveData;
    }
}

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/viewmodel/AnswerViewModel.kt
package com.josycom.mayorjay.flowoverstack.ui.viewmodel

import com.josycom.mayorjay.flowoverstack.data.AnswerRepository
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import androidx.paging.PagingData
import androidx.paging.cachedIn
import com.josycom.mayorjay.flowoverstack.model.Answer
import kotlinx.coroutines.flow.Flow

class AnswerViewModel(answerRepository: AnswerRepository, questionId: Int, order: String, sortCondition: String, site: String, filter: String, siteKey: String) : ViewModel() {

    val answerDataFlow: Flow<PagingData<Answer>>?

    init {
        answerRepository.init(questionId, order, sortCondition, site, filter, siteKey)
        answerDataFlow = answerRepository.answerDataFlow?.cachedIn(viewModelScope)
    }
}